src/io/hdf_dataproxy.h

```diff
@@ -47,6 +48,45 @@ inline bool get_space(hid_t grp, const std::string& aname, int rank, hsize_t* di
   return thesame;
 }
 
+/** free function to go from spec to dimensionality of memory space and data space
```

Comment:
I do see the need of query dataset on the file via hdf_archive. it is currently only allowed inside the class.